packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,206 @@
+
+    if (coordinates) {
+        ensureUniqueContentDivId(core);
+        table.id = ensureUniqueId(table, doc);
+        const ranges = select(table, coordinates, core);
+        return {
+            type: SelectionRangeTypes.TableSelection,
+            ranges,
+            table,
+            areAllCollapsed: ranges.filter(range => range?.collapsed).length == ranges.length,
+            coordinates,
+        };
+    }
+
+    unselect();
+
+    if (!coordinates && styleElement) {
+        styleElement.parentElement.removeChild(styleElement);
+        styleElement = null;
+    }
+    return null;
+};
+
+function buildCss(
+    table: HTMLTableElement,
+    coordinates: TableSelection,
+    contentDivSelector: string
+): { css: string; ranges: Range[] } {
+    coordinates = normalizeTableSelection(coordinates);
+    const tr1 = coordinates.firstCell.y + 1;
+    const td1 = coordinates.firstCell.x + 1;
+    const tr2 = coordinates.lastCell.y + 1;
+    const td2 = coordinates.lastCell.x + 1;
+    const ranges: Range[] = [];
+    const tbodySelector = table.querySelector('tbody') ? '>tbody' : '';
+    let firstSelected: HTMLTableCellElement | null = null;
+    let lastSelected: HTMLTableCellElement | null = null;
+
+    let css = '';
+
+    let isFirst = true;
+    for (let i = tr1; i <= tr2; i++) {
+        firstSelected = null;
+        lastSelected = null;
+
+        for (let j = td1; j <= td2; j++) {
+            if (isFirst) {
+                isFirst = false;
+            } else {
+                css += ',';
+            }
+
+            let selector =
+                contentDivSelector +
+                ' #' +
+                table.id +
+                tbodySelector +
+                ' >tr:nth-child(' +
+                i +
+                ')>td:nth-child(' +
+                j +
+                ')';
+            if (table.querySelector(selector)) {
+                css += selector;
+            } else {
+                selector = selector.replace('td', 'th');
+                if (table.querySelector(selector)) {
+                    css += selector;
+                }
+            }
+        }
+
+        const rowSelector = 'tr:nth-child(' + i + ')>';
+        const firstSelector = `${rowSelector}td:nth-child(${td1}),${rowSelector}th:nth-child(${td1})`;
+        const lastSelector = `${rowSelector}td:nth-child(${td2}),${rowSelector}th:nth-child(${td2})`;
+
+        firstSelected = table.querySelector(firstSelector);
+        lastSelected = table.querySelector(lastSelector)!;
+        if (firstSelected && lastSelected) {
+            const rowRange = new Range();
+            rowRange.setStartBefore(firstSelected);
+            rowRange.setEndAfter(lastSelected);
+            ranges.push(rowRange);
+        }
+    }
+
+    css += '{background-color: rgba(198,198,198,0.7) !important;}';
+
+    return { css, ranges };
+}
+
+function select(table: HTMLTableElement, coordinates: TableSelection, core: EditorCore): Range[] {
```

Comment:
suggest make core as the first parameter, to follow the same pattern with other functions